### ISSUE TITLE
--show-error-end does not show the error end in Python 3.7

### ISSUE THREAD

**Issue by Richardk2n** [python/mypy#15281]
**Bug Report**

`--show-error-end` does not show the error end in Python 3.7

**To Reproduce**

Run `mypy pythonFileWithError.py --show-error-end` with Python 3.7 with the file containing the following.

```python
{}.append(3)
```

**Expected Behavior**

Should output (and does in newer Python versions):
```
pythonFileWithError.py:1:1:1:9: error: "Dict[<nothing>, <nothing>]" has no attribute "append"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

Does output:
```
pythonFileWithError.py:1:1:1:1: error: "Dict[<nothing>, <nothing>]" has no attribute "append"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.3.0
- Mypy command-line flags: --show-error-end
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.7.13

**Comment by hauntsaninja** [comment]
Unfortunately, the Python AST does not expose `end_col_offset` on Python 3.7 and older, so this is a won't fix. Especially given that end of life for 3.7 is in about a month.

**Comment by hauntsaninja** [comment]
If you think this is likely to trip people up, we can add a note to the documentation. Although again mypy's probably going to drop support for 3.7 sometime this year

**Comment by Richardk2n** [comment]
Thank, you. I kind of already had a feeling, something like that would be the answer.
I do not expect a fix as I am well aware of the eol of 3.7, just wanted to confirm the error is not on my end.
As I tripped over the fact, that there was no note to be found, I would appreciate a note in the documentation.
